### PR TITLE
Avoid unnecessary `return await`

### PR DESCRIPTION
**Pull request by fregante**
I was making some changes to the code and found myself having to fix a few Promise-related issues, so I thought I'd start pushing some fixes for it, one PR per kind. These changes help us avoid callback hell and recover errors that have been falling through the cracks.

---

This minor change is the application of [no-return-await](https://eslint.org/docs/rules/no-return-await) eslint rule. Nothing changed here except removing some code.

In short, `return await` is only ever needed if you want to await a promise in the current `try/catch` block.

For example, given that `bar` is an async function:
```js
async function foo() {
    try {
        return await bar(); // The error from `bar` is caught inside foo
    } catch (error) {}
}
```

```js
async function foo() {
    try {
        return bar(); // The error from `bar` is NOT caught inside foo
    } catch (error) {}
}
```

### CODE REVIEW

File: src/background/logging.ts [hunk 1]
```diff
@@ -315,13 +315,13 @@ export async function _setLoggingConfig(config: LoggingConfig): Promise<void> {
 export const getLoggingConfig = liftBackground(
   "GET_LOGGING_CONFIG",
   async () => {
-    return await _getLoggingConfig();
+    return _getLoggingConfig();
   }
```
Comment (fregante) [inline src/background/logging.ts:319]: An additional fix that this change highlights, would be to drop the wrapper function in many cases, but this might change the behavior slightly.
```suggestion
  _getLoggingConfig
```

But it can be made safe again with

```suggestion
  unary(_getLoggingConfig)
```

Want this change too, where possible?

File: src/background/logging.ts [hunk 2]
```diff
@@ -315,13 +315,13 @@ export async function _setLoggingConfig(config: LoggingConfig): Promise<void> {
...
 );
 
 export const setLoggingConfig = liftBackground(
   "SET_LOGGING_CONFIG",
   async (config: LoggingConfig) => {
-    return await _setLoggingConfig(config);
+    return _setLoggingConfig(config);
   }
```
Comment (fregante) [inline src/background/logging.ts:326]: An additional fix that this change highlights, would be to drop the wrapper function in many cases, but this might change the behavior slightly.
```suggestion
  _getLoggingConfig
```

But it can be made safe again with

```suggestion
  ary(_getLoggingConfig, 1)
```

Want this change too, where possible? In many cases this avoids having to duplicate types in this wrapper function

File: src/nativeEditor/dynamic.ts [hunk 3]
```diff
@@ -84,7 +84,7 @@ async function buildSingleReader(config: ReaderLike): Promise<IReader> {
   } else if (isCustomReader(config)) {
     return readerFactory(config);
   } else {
-    return (await blockRegistry.lookup(config.metadata.id)) as IReader;
+    return blockRegistry.lookup(config.metadata.id) as Promise<IReader>;
```
Comment (fregante) [inline src/nativeEditor/dynamic.ts:87]: The second commit unwraps this second style of `return await`, which made sense as a way to do assertions. It doesn't change the length but it just moves the whole burden to the right side of the assertion.

I hope to drop these assertions at some point too, if possible.

File: src/background/logging.ts (hunk 2 above)
Comment (twschiller) [reply src/background/logging.ts:326]: I have a slight readability preference for `unary` vs. `ary`

I think I prefer the more verbose arrow function since it's less error prone if you forget to wrap with unary. But Typescript would potentially catch most errors of forgetting to wrap in unary?

File: src/nativeEditor/dynamic.ts (hunk 3 above)
Comment (twschiller) [reply src/nativeEditor/dynamic.ts:87]: I agree those assertions are ugly/error-prone. The block registry is hard to statically type with Typescript since the types are dynamic based on the config. 

Basically we'd need to add some type information that's then enforced dynamically by the lookup method

**Review by twschiller — APPROVED** [review]

File: src/nativeEditor/dynamic.ts (hunk 3 above)
Comment (fregante) [reply src/nativeEditor/dynamic.ts:87]: > fall under a different PR though

Yes definitely. I'm grouping these PRs by type of change so that generally you can look at a file and understand how the rest of the files are modified. 

In these PRs I'm generally leaving the logic as it was before, maybe with a TODO comment for possible improvements.

File: src/background/logging.ts (hunk 2 above)
Comment (fregante) [reply src/background/logging.ts:326]: Naked function references are probably unsafe, as I learned recently in a discussion with Addy Osmani if I remember correctly. There's a related rule for this, but it's only for arrays: https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-array-callback-reference.md

We can circle back to these once I start looking at the messaging
